proposed/2022/package-scoring-v1.md

```diff
@@ -0,0 +1,219 @@
+
+Package scoring will contain four unique categories to start off with. The categories are the following:
+
+- Popularity - How popular a package is & recognized in the ecosystem.
+- Quality - The completeness of a package following best practices & providing documentation.
+- Maintenance - The state of maintenance for a package based on it's update cadence.
+- Security - The sense of trust based on a package being free of known security vulnerabilities, license risk, and current supply chain risk.
+
+Each category has the potential to total up to 100 points.
+
+- Categories in which the package is **in control** can always reach the maximum amount of points.
+- Categories in which the package is **not in control** will be determined by the state of the ecosystem.
+
+EX: A package is in control of providing ample metadata to ensure the quality of the package reaches the full score. A package is not in control of how popular it may be however.
+
+To scale package scoring for the future, each score will be comprised of an analysis of package performance and potential package/dependency issues. Each of these issues will be categorized into one of these four categories and empower the author and consumer to resolve these issues to the best of their ability and control.
+
+#### Issues
+
+As a means to scale package scoring to it's potential through continuous iteration, each category will have its own set of issues that a user can be aware of when browsing through the package. Keeping the same idea of **being in control** or **not in control** also applies to package issues.
+
+For issues which can be addressed, an issue will include an empowering message regarding how one can take action to help address the issue.
+
+EX: A package might be missing a README which a developer may not be able to easily get started with the package. They might be able to then contribute a new issue or even the inclusion of the README file to the NuGet package if it's open source.
+
+**Issue Examples:**
+
+- Missing dependency
+- Missing README
+- No example
+- Not enough public API documentation
+- Bad semver
+- Not v1+
+- No website
+- No repository
+- Unmaintained
+- No bug tracker
+- Has CVE (Critical, High, etc)
+- Deprecated license
+- Missing license
+- Non SPDX license
+- Unsafe copyright
+- Deprecated
+- Empty package
+- etc
+
+#### Popularity
+
+Popularity is the more unique of the other three categories which are more issue based. Popularity is calculated as how many projects depend on a package over the past 30 days. This is scored as a percentile up to 100% (most used) to 0% (least used).
+
+Since this is not possible with current data pipelines, we will work on providing such an experience in the future and use two proxy values such as total weekly downloads and total count of packages depending on the package to be scored up to a percentile of up to 100%.
+
+- Total Weekly Downloads
```

Comment:
Why weekly? Because during XMAS down time someone run bunch of bots to generate 10k downloads for own package, then it it's placed at higher in ranking it becomes it gets more downloads in following weeks. To reduce weight of this kind of spike in finance use [moving average](https://en.wikipedia.org/wiki/Moving_average). Actually, implementing moving average is complicated.